recipes/openjpeg/all/conanfile.py

```diff
@@ -8,102 +8,94 @@ class OpenjpegConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     description = "OpenJPEG is an open-source JPEG 2000 codec written in C language."
     topics = ("conan", "jpeg2000", "jp2", "openjpeg", "image", "multimedia", "format", "graphics")
-    options = {"shared": [True, False], "build_codec": [True, False], "fPIC": [True, False]}
-    default_options = {'shared': False, 'build_codec': True, 'fPIC': True}
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package"
     homepage = "https://github.com/uclouvain/openjpeg"
     license = "BSD 2-Clause"
-    exports_sources = ["CMakeLists.txt"]
 
-    _source_subfolder = "source_subfolder"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_codec": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_codec": False
+    }
+
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
+    def package_id(self):
+        self.info.options.build_codec # not used for the moment
+
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
     def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions['BUILD_SHARED_LIBS'] = self.options.shared
-        cmake.definitions['BUILD_STATIC_LIBS'] = not self.options.shared
-        cmake.definitions['BUILD_PKGCONFIG_FILES'] = False
-        cmake.definitions['CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP'] = True
-        cmake.definitions['BUILD_CODEC'] = False
-
-        cmake.configure()
-        return cmake
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = True
+        self._cmake.definitions["BUILD_DOC"] = False
+        self._cmake.definitions["BUILD_STATIC_LIBS"] = not self.options.shared
+        self._cmake.definitions["BUILD_LUTS_GENERATOR"] = False
+        self._cmake.definitions["BUILD_CODEC"] = False
```

Comment:
Yes, correct.
Actually, `build_codec`recipe's option has never been mapped to `BUILD_CODEC` cmake option, before or after https://github.com/conan-io/conan-center-index/commit/915befb7b30a108a92cc941d84cb12911849ec6c.
I don't think consumers has really noticed this modification, because bin folder was not added in PATH env var.

Bu yes, to build codecs, cmake_find_package and patches are required, except this one which is useless (at least since 2.3.1):

```python
        # fix installing libs when only shared or static library built
        tools.replace_in_file(os.path.join(self._source_subfolder, 'src', 'lib', 'openjp2', 'CMakeLists.txt'),
                              'add_library(${OPENJPEG_LIBRARY_NAME} ${OPENJPEG_SRCS})',
                              'add_library(${OPENJPEG_LIBRARY_NAME} ${OPENJPEG_SRCS})\n'
                              'set(INSTALL_LIBS ${OPENJPEG_LIBRARY_NAME})')
```

And I don't understand why `libpng` was disabled.

Reply:
> Actually, build_codecrecipe's option has never been mapped to BUILD_CODEC CMakeLists option, before or after 915befb.
>  I don't think consumers has really noticed this modification, because bin folder was not added in PATH env var.

If so, it is a safe change.

> And I didn't understand why libpng was disabled.

Do not remember too. I had added It in an initial commit. Anyway it is only used for codecs